packages/bff/src/services/agreementService.ts

```diff
@@ -57,3 +146,240 @@ export const getLatestAgreement = async (
     }
   })[0];
 };
+
+function isUpgradable(
+  descriptor: catalogApi.EServiceDescriptor,
+  agreement: agreementApi.Agreement,
+  descriptors: catalogApi.EServiceDescriptor[]
+): boolean {
+  return descriptors
+    .filter((d) => parseInt(d.version, 10) > parseInt(descriptor.version, 10))
+    .some(
+      (d) =>
+        (d.state === "PUBLISHED" || d.state === "SUSPENDED") &&
+        (agreement.state === "ACTIVE" || agreement.state === "SUSPENDED")
+    );
+}
+
+async function enrichAgreement(
```

Comment:
I know it's Scala naming, but `enrich` and `enhance` are not so meaningful to understand how they differ 🥲  In other services we always used the term enhance, maybe here we could call these two `enhanceAgreementForGetAgreements` and `enhanceAgreementForGetAgreementById`?

Reply:
Since enhanceAgreement will be used in other routes as well, what about `enhanceAgreementOverview` for the less detailed one and `enhanceAgreementDetailed`?

Reply:
Works!